### PR TITLE
Exo role group

### PR DESCRIPTION
**Pull request by afloca**
#### Pull Request (PR) description

Initial release, Create Exchange Role Group with Roles and Members

### CODE REVIEW

**Comment by afloca** [comment]
@microsoft-github-policy-service agree [company="swisscom"]

________________________________

**Comment by afloca** [comment]
@microsoft-github-policy-service agree company="swisscom"

________________________________

**Comment by andikrueger** [comment]
Could you please add a new entry within UNRELEASED section of the changelog to outline the new resource.

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXORoleGroup/MSFT_EXORoleGroup.psm1 [hunk 1]
```diff
@@ -0,0 +1,452 @@
+function Get-TargetResource {
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXORoleGroup/MSFT_EXORoleGroup.psm1:1]: The code style you are using does not match the style used within M365DSC. Specifically this one: "One Newline Before Braces" here (https://dsccommunity.org/styleguidelines/whitespace/)

Are you using Visual Studio Code and have the PowerShell Extension installed, you can then run the formatter by pressing Alt-Shift-F which will apply the formatting used by the project.

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXORoleGroup/MSFT_EXORoleGroup.psm1 [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+    # Remove Description Parameter if null or Empty as the creation fails with $null parameter
+    if ([System.String]::IsNullOrEmpty($Description)) {
+        $NewRoleGroupParams.Remove('Description') | Out-Null
+    }
+    # CASE: Role Group doesn't exist but should;
+    if ($Ensure -eq 'Present' -and $currentRoleGroupConfig.Ensure -eq 'Absent') {
+        Write-Verbose -Message "Role Group '$($Name)' does not exist but it should. Create and configure it."
+        # Create Role Group
+        New-RoleGroup @NewRoleGroupParams
+
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXORoleGroup/MSFT_EXORoleGroup.psm1:216]: Please remove this empty line

File: Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            Mock -CommandName Get-PSSession -MockWith {
+            }
+
+            Mock -CommandName Remove-PSSession -MockWith {
+            }
+
+            # Mock Write-Host to hide output during the tests
+            Mock -CommandName Write-Host -MockWith {
+            }
+            Function get-RoleGroup{
```
Comment (ykuijs) [inline Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1:45]: Shy are you adding these functions here?

My guess is that you want to fake these cmdlets, but those so called "stubs" should be placed in this file:
https://github.com/microsoft/Microsoft365DSC/blob/Dev/Tests/Unit/Stubs/Microsoft365.psm1

That way you can Mock them later in the tests.

File: Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                    return $null
+                }
+                Test-TargetResource @testParams | Should -Be $false
+            }
+
+            It 'Should call the Set method' {
+                Mock -CommandName New-RoleGroup -MockWith {
+                    return $null
+                }
+                Set-TargetResource @testParams
```
Comment (ykuijs) [inline Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1:92]: I believe you should check for something here, like Should -Invoke New-RoleGroup. Without adding a check for something specific, it is possible that the code returns something totally different and still succeed.

More info: https://pester.dev/docs/commands/Should#example-1

File: Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                }
+
+            }
+
+            It 'Should return false from the Test method' {
+                Test-TargetResource @testParams | Should -Be $false
+            }
+
+            It 'Should call the Set method' {
+                Set-TargetResource @testParams
```
Comment (ykuijs) [inline Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1:147]: Same as before, the test should actually test something.

**Review by ykuijs — CHANGES_REQUESTED** [review]
Great work! Thanks for this PR. I have added a few comments and could you also please add an example for the resource?

**Comment by ykuijs** [comment]
@afloca, could you please incorporate the review comments and update your branch with Dev? Maybe we can include your PR in tomorrows release.

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXORoleGroup/MSFT_EXORoleGroup.psm1 (hunk 1 above)
Comment (afloca) [reply Modules/Microsoft365DSC/DSCResources/MSFT_EXORoleGroup/MSFT_EXORoleGroup.psm1:1]: for some reason it is exactly the VS Code PS extension removing the new lines before braces when runing the formatter.

File: Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1 (hunk 3 above)
Comment (afloca) [reply Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1]: added the Functions as stubs

**Pull request closed without merging**